### PR TITLE
Searching for PID's of running python processes

### PR DESCRIPTION
**Pull request by vvaryshchuk**

### CODE REVIEW

File: vasyl_varyshchuk/Week_2/task_3/search_python.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Search python processes
+
+key_word=$1
+
+ps -efl | grep "$key_word" | awk  '!/bash|grep/ {print $4}'
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_3/search_python.sh]: Could you please describe in docstring how it works?

File: vasyl_varyshchuk/Week_2/task_3/show_pids.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+"""
+
+import argparse
+from subprocess import call
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--search', help='search for python process')
+args = parser.parse_args()
+
+argument = str(args.search)
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_3/show_pids.py]: What is `args.search`?

File: vasyl_varyshchuk/Week_2/task_3/search_python.sh (hunk 1 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_3/search_python.sh]: Ok, description was added to search_python.sh script.

File: vasyl_varyshchuk/Week_2/task_3/show_pids.py (hunk 2 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_3/show_pids.py]: In my case, args.search returns command line argument for flag --search.

File: vasyl_varyshchuk/Week_2/task_3/search_python.sh (hunk 1 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_3/search_python.sh]: The main reason to use it in this command is to filter processes which in fact aren't a python processes but contains 'python' in the output fields of ps command. For example, in the following scripts, I will display an additional column that is responsible for the name of the process (СMD in ps command output).

First I run this script in the terminal without regex pattern in awk:

    ps -efl | grep python | awk '{print $4 "\t" $15}'

Output:
    23342	    /usr/bin/python3.5
    23349	    /usr/bin/python3.5
    23356	    grep

And then with regex pattern in awk:

    ps -efl | grep python | awk  '!/bash|grep/ {print $4 "\t" $15}'

Output:
    23342	    /usr/bin/python3.5
    23349	    /usr/bin/python3.5

File: vasyl_varyshchuk/Week_2/task_3/search_python.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# This script searches for running processes key_word pattern passed from command line
+# ps command in combination with -efl options creates a table with information for each processes.
+# After that, grep finds lines containing 'Skey_word' pattern, whitch is passed from command line.
```
[the remark this answers is not in the document]
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_3/search_python.sh]: I corrected mistakes and committed my changes

File: vasyl_varyshchuk/Week_2/task_3/search_python.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+# Finally awk prints colomn which corresponds to PID's, excluding lines which conatains 'bash' or
```
[the remark this answers is not in the document]
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_3/search_python.sh]: I corrected mistakes and committed my changes

File: vasyl_varyshchuk/Week_2/task_3/show_pids.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+"""
+This module calls a bash script that searches for running
+python processes and displays PID's of these processes
+"""
+
+import argparse
+from subprocess import call
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--search', help='search for python process')
```
[the remark this answers is not in the document]
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_3/show_pids.py]: Thanks for your advice, I added short alias in my code

File: vasyl_varyshchuk/Week_2/task_3/search_python.sh (hunk 1 above)
Comment (pivancy) [reply vasyl_varyshchuk/Week_2/task_3/search_python.sh]: The easier way is to do something like:

    ps auxww | grep [p]ython

It shows exactly python processes without any grep, etc.

**Comment by pivancy** [comment]
:+1:

File: vasyl_varyshchuk/Week_2/task_3/search_python.sh (hunk 1 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_3/search_python.sh]: Very nice example. Thanks.

**Pull request merged**